Application.mk

```diff
@@ -54,10 +54,13 @@ endif
 ifeq ($(BUILD_MODULE),y)
   # Revisit: This only works for gcc and clang.
   # Do other compilers have similar?
-  COMPILER_RT_LIB = $(shell $(CC) $(ARCHCPUFLAGS) --print-libgcc-file-name)
-  ifneq ($(COMPILER_RT_LIB),)
-    LDLIBS += $(COMPILER_RT_LIB)
+  LIBGCC = $(shell $(CC) $(ARCHCPUFLAGS) --print-libgcc-file-name)
```

Comment:
i named the variable COMPILER_RT_LIB because it isn't always libgcc.
```
spacetanuki% cc --print-libgcc-file-name
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/13.1.6/lib/darwin/libclang_rt.builtins-x86_64.a
spacetanuki% 
```

Reply:
@anchao do you know why `--print-libgcc-file-name` fails? is it because of `ARCHCPUFLAGS` do not contain a proper flags?
I mean that I'm not sure if we need fallback path. Do you have an example output of `$(shell $(CC) $(ARCHCPUFLAGS) --print-libgcc-file-name)` that meets  `ifeq ($(wildcard $(LIBGCC)),)` condition?

Reply:
From the OS part we have
```
EXTRA_LIBS += ${wildcard ${shell $(CC) $(ARCHCPUFLAGS) --print-libgcc-file-name}}

ifneq ($(CONFIG_LIBM),y)
  EXTRA_LIBS += ${wildcard ${shell $(CC) $(ARCHCPUFLAGS) --print-file-name=libm.a}}
endif

ifeq ($(CONFIG_LIBSUPCXX),y)
  EXTRA_LIBS += ${wildcard ${shell $(CC) $(ARCHCPUFLAGS) --print-file-name=libsupc++.a}}
endif
```
so there is no fallback

Reply:
Ok. Just saw https://github.com/apache/incubator-nuttx/pull/6076 and that you are modifying OS part as well

Reply:
Ok, I will restore this naming

Reply:
> @anchao do you know why --print-libgcc-file-name fails? is it because of ARCHCPUFLAGS do not contain a proper flags?
I mean that I'm not sure if we need fallback path. Do you have an example output of $(shell $(CC) $(ARCHCPUFLAGS) --print-libgcc-file-name) that meets ifeq ($(wildcard $(LIBGCC)),) condition?


I think this should be a bug of arm clang, "--print-libgcc-file-name" gives the correct library name but with incorrect path,
this library does not exist in the toolchain,  so if "wildcard" fails,  we should try the correct library name with "--print-file-name "

--print-libgcc-file-name:
```
$ clang  --config armv8m.main_hard_fp_nosys --target=armv8m.main-none-eabi -mfloat-abi=hard  -march=armv8m.main+fp --print-libgcc-file-name

/home/archer/code/l61/ap/prebuilts/clang/linux/arm/lib/clang/13.0.0/lib/baremetal/libclang_rt.builtins-armv8m.main.a
```

Path is wrong:
/home/archer/code/l61/ap/prebuilts/clang/linux/arm/lib/clang/13.0.0/lib/baremetal/libclang_rt.builtins-armv8m.main.a

--print-file-name:
```
$ clang  --config armv8m.main_hard_fp_nosys --target=armv8m.main-none-eabi -mfloat-abi=hard  -march=armv8m.main+fp --print-file-name libclang_rt.builtins-armv8m.main.a

/home/archer/code/l61/ap/prebuilts/clang/linux/arm/bin/../lib/clang-runtimes/armv8m.main_hard_fp/lib/libclang_rt.builtins-armv8m.main.a
```

Reply:
Seems like this has to be reported to LLVM because it is a bug. The workaround is fine.